schemast/annotation.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2019-present Facebook
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schemast
+
+import (
+	"fmt"
+	"go/ast"
+	"sort"
+
+	"entgo.io/contrib/entproto"
+	"entgo.io/ent/dialect/entsql"
+	"entgo.io/ent/schema"
+	"github.com/mitchellh/mapstructure"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+type AnnotNotSupportedErr struct {
+	annot schema.Annotation
+}
```

Comment:
```suggestion
type UnsupportedAnnotError struct {
	annot schema.Annotation
}
```
See the [UnsupportedTypeError](https://golang.org/pkg/encoding/json/#UnsupportedTypeError) as an example. 

I'd keep the full `Annotation` word, even though it's longer.